### ISSUE TITLE
Use Admin Framework from CounterStrikeSharp

### ISSUE THREAD

**Issue by kus** [shobhit-pathak/MatchZy#11]
CounterStrikeSharp has introduced an [Admin Framework](https://docs.cssharp.dev/features/admin-framework/) which supports admins and flags similar to SourceMod, removing the need to have an admin file for every plugin.

Could you please update your plugin to use this new system so we don't have to have an individual admin file for your plugin.

**Comment by shobhit-pathak** [comment]
Hey!
I will surely update my plugin to use the new admin system, I'll check what all changes I have to make because there are already a lot of admin commands

**Comment by drakin001** [comment]
+1 this would be great!

**Comment by crashzk** [comment]
@shobhit-pathak I believe we can use the whitelist system with CSSharp too, right?

In other words, do players who have the CSSharp admin flag automatically enter the Whitelist?

**Comment by shobhit-pathak** [comment]
@crashzk Yes I think this can be done, but I was still not able to explore the admin system of CSSharp, because there are already a lot of admin commands and checks in MatchZy which I will have to update :P

**Comment by ChocOranger** [comment]
Definitely a big +1 for me on more granular admin settings.

I'd bite the bullet and make this a priority because the more commands you add the longer it's going to take to implement.

And if I could add a suggestion:

.map and .prac and probably .exitprac should be grouped together under the map change admin group (@css/changemap)

All other match commands should be separated into cfg change (@css/config)

This would mean that more junior members of a team could join and change the map and practise without having the ability to affect matches.

**Comment by shobhit-pathak** [comment]
Yeah that makes sense, I think at this point I should start focusing on using CSSharp's admin framework and pause the other developments for a while. Because as you said, the more commands I add, the longer its going to take to implement CSSharp's admin system.

**Comment by kus** [comment]
I know CS2Rcon has a lot less functionality than your plugin, but this is the [commit](https://github.com/LordFetznschaedl/CS2Rcon/commit/be0cef22665d7e6f5b4882dc02d48b79ce84d046) where they changed from their own admin system to CSSharp's and as you can see it cleaned up a lot of their code!

**Comment by crashzk** [comment]
@shobhit-pathak first of all, thank you for this resource, testing it here now.

Just an observation, I noticed in this case that it would be the `@css/generic` flag for all commands that need to be admin, right?

If so, in the future I believe that an improvement to this feature would be interesting, that is, each admin command has a flag that we can configure which one to use.

Example:
- If I want to leave the `.prac`, `.exitprac` commands only for admin with the `@css/root` flag.
- `.rcon` command only for those who have the `@css/rcon` flag

And so on.

**Comment by ChocOranger** [comment]
@crashzk the @css/generic flag is needed to make a player an admin. Without it other flags won't work.

**Comment by ChocOranger** [comment]
@crashzk forget what I said above, that's how source mod works.
The @css/generic sets a player as admin. No other flags do anything.

**Comment by ChocOranger** [comment]
@shobhit-pathak I'm not sure this should be closed.

Sure it allows admins to be added for CSSharp but it's just an all or nothing approach. All admins have full access to all admin protected commands.
That's not really how the CSSharp system was designed to be used.
